frontend/src/app/sessions/new-readonly-session-dialog/new-readonly-model-options.component.ts

```diff
@@ -0,0 +1,121 @@
+/*
+ * SPDX-FileCopyrightText: Copyright DB Netz AG and the capella-collab-manager contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Component, Input, OnInit } from '@angular/core';
+import {
+  FormControl,
+  FormGroup,
+  Validators,
+  ValidationErrors,
+  ValidatorFn,
+  AbstractControl,
+} from '@angular/forms';
+
+import { GetGitModel } from 'src/app/projects/project-detail/model-overview/model-detail/git-model.service';
+import { Revisions, GitService } from 'src/app/services/git/git.service';
+import { Model } from 'src/app/services/model/model.service';
+
+export type ModelOptions = {
+  model: Model;
+  primary_git_model: GetGitModel;
+  include: boolean;
+  revision: string;
+  deepClone: boolean;
+};
+
+@Component({
+  selector: 'new-readonly-model-options',
+  templateUrl: './new-readonly-model-options.component.html',
+  styleUrls: ['./new-readonly-model-options.component.css'],
+})
+export class NewReadonlyModelOptionsComponent implements OnInit {
+  @Input() projectSlug = '';
+  @Input() modelOptions: ModelOptions | undefined;
+
+  constructor(private gitService: GitService) {}
+
+  public filteredRevisions: Revisions = { branches: [], tags: [] };
+
+  public availableRevisions: Revisions | undefined = {
+    branches: [],
+    tags: [],
+  };
```

Comment:
What is the difference between `filteredRevisions` and `availableRevisions`?

Reply:
Available revisions should be all revisions from the repo, filtered revisions are the ones that match with the text entered in the input field so far, only that part has not been implemented.

Reply:
Do you plan to implement it? If not, we can also use a `MatSelect` for the selection of the branch/tag.

Reply:
Implemented